src/Gremlin.Net.CosmosDb/Serialization/AutoConnector.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Gremlin.Net.CosmosDb.Structure;
+
+namespace Gremlin.Net.CosmosDb.Serialization
+{
+    internal class AutoConnector : IVertexConnector, IEdgeConnector
+    {
+        public bool CanConnectEdge(IVertex vertex, IEdge edge, PropertyInfo property) => true;
+        public bool CanConnectVertex(IEdge edge, IVertex vertex) => true;
+
+        public bool ConnectEdge(IVertex vertex, IEdge edge, PropertyInfo property)
+        {
+            if (!(property.GetValue(vertex) is IEdge existingEdge))
+            {
+                property.SetValue(vertex, edge);
+            }
+            else
+            {
+                var vertexProps = edge.GetType().GetProperties().Where(p => TypeCache.IVertex.IsAssignableFrom(TypeHelper.UnderlyingType(p.PropertyType)));
+
+                foreach (var prop in vertexProps)
+                {
+                    if (prop.GetValue(edge) is object newEdgeVertex)
```

Comment:
NIT: Using a switch here may be a bit cleaner than all the if/is checks you're doing:
```c#
switch (prop.GetValue(edge))
{
    case IVertex v:
        ConnectVertex(existingEdge, v);
        break;
    case IEnumerable vEnum:
        foreach (var vert in vEnum.OfType<IVertex>())
        {
            ConnectVertex(existingEdge, vert);
        }
        break;
}
```